src/frameworks/utils.ts

```diff
@@ -63,20 +63,47 @@ export async function warnIfCustomBuildScript(
   }
 }
 
-function proxyResponse(original: ServerResponse, next: () => void) {
-  return (response: IncomingMessage | ServerResponse) => {
-    const { statusCode, statusMessage } = response;
-    if (!statusCode) {
-      original.end();
-      return;
-    }
-    if (statusCode === 404) {
-      return next();
-    }
-    const headers = "getHeaders" in response ? response.getHeaders() : response.headers;
-    original.writeHead(statusCode, statusMessage, headers);
-    response.pipe(original);
-  };
+export function proxyResponse(req: IncomingMessage, res: ServerResponse, next: () => void) {
```

Comment:
I think it would be great if you could add a brief description to what this function is doing now, I'm glad it works but I have no idea why it fixes the errors :joy:

Reply:
I added some comments.